File: src/hunt_optimizer/dto/drops.ts
export type EnemyDropDto = {
    difficulty: string;
    episode: number;
    section_id: string;
    enemy: string;
    item_type_id: number;
    drop_rate: number;
    rare_rate: number;
};

export type BoxDropDto = {
    difficulty: string;
    episode: number;
    section_id: string;
    area_id: number;
    item_type_id: number;
    drop_rate: number;
};

File: web/shared/src/commonMain/kotlin/world/phantasmal/web/shared/dto/Drops.kt
package world.phantasmal.web.shared.dto

import kotlinx.serialization.Serializable
import world.phantasmal.lib.Episode
import world.phantasmal.lib.fileFormats.quest.NpcType

@Serializable
class EnemyDrop(
    val difficulty: Difficulty,
    val episode: Episode,
    val sectionId: SectionId,
    val enemy: NpcType,
    val itemTypeId: Int,
    val dropRate: Double,
    val rareRate: Double,
)

@Serializable
class BoxDrop(
    val difficulty: Difficulty,
    val episode: Episode,
    val sectionId: SectionId,
    val areaId: Int,
    val itemTypeId: Int,
    val dropRate: Double,
)
